### PR TITLE
Class based configs #133

### PR DESCRIPTION
**Pull request by annawoodard**
This should be considered in conjunction with Parsl/parsl#335, which moves from using dictionary configs for the DFK to a `Config` class. Everything before 6cf6699 are non-essential changes; 6cf6699 and 4af3929 implement the required changes for Parsl/parsl#335. Note that Parsl/parsl#335 depends on this PR, so this should be reviewed and a point release made before reviewing Parsl/parsl#335. Please see Parsl/parsl#335 for related info.

### CODE REVIEW

File: libsubmit/providers/condor/condor.py [hunk 1]
```diff
@@ -5,7 +5,6 @@
 from string import Template
```
Comment (yadudoc) [inline libsubmit/providers/condor/condor.py]: We stopped here at 1:11. resume review from here.

File: libsubmit/providers/condor/condor.py [hunk 2]
```diff
@@ -315,7 +252,7 @@ def submit(self, command, blocksize, job_name="parsl.auto"):
         channel_script_path = self.channel.push_file(script_path, self.channel.script_dir)
 
         cmd = "condor_submit {0}".format(channel_script_path)
-        retcode, stdout, stderr = self.channel.execute_wait(cmd, 3, envs=self.config['execution']['environment'])
+        retcode, stdout, stderr = self.channel.execute_wait(cmd, 3)
```
Comment (yadudoc) [inline libsubmit/providers/condor/condor.py]: I think we might need these envs set.

File: libsubmit/providers/local/local.py [hunk 3]
```diff
@@ -214,7 +163,7 @@ def submit(self, command, blocksize, job_name="parsl.auto"):
         script_path = "{0}/{1}.sh".format(self.script_dir, job_name)
         script_path = os.path.abspath(script_path)
 
-        wrap_command = self.launcher(command, taskBlocks=self.taskBlocks)
+        wrap_command = launchers.get(self.launcher, None)(command, self.tasks_per_node * self.nodes_per_block)
```
Comment (yadudoc) [inline libsubmit/providers/local/local.py]: If self.launcher is not set, launchers.get will return None which is not a function. We need to fix this.

File: libsubmit/providers/condor/condor.py (hunk 2 above)
Comment (annawoodard) [reply libsubmit/providers/condor/condor.py]: Looking at this again, I see that there are two sets of envs: what is set in the constructor are envs to be set for the task, these are set before running commands in the channel. I think it makes more sense to set these in the channel, by adding an `envs` keyword arg to the channel (which brings the interface for `SSHChannel`, `SSHInteractiveLoginChannel`, and `LocalChannel` into sync). I've done this in f877068. Let me know if you have objections and I can change it back, or add a second, channel-specific `envs` keyword argument to the `Condor` provider.

Comment (yadudoc) [reply libsubmit/providers/condor/condor.py]: Got it.

**Pull request merged**